src/liboslexec/llvm_util.cpp

```diff
@@ -825,16 +796,67 @@ LLVM_Util::make_jit_execengine (std::string *err)
 
     engine_builder.setEngineKind (llvm::EngineKind::JIT);
     engine_builder.setErrorStr (err);
+    //engine_builder.setRelocationModel(llvm::Reloc::PIC_);
+    //engine_builder.setCodeModel(llvm::CodeModel::Default);
+    engine_builder.setVerifyModules(true);
 
     // We are actually holding a LLVMMemoryManager
     engine_builder.setMCJITMemoryManager (std::unique_ptr<llvm::RTDyldMemoryManager>
         (new MemoryManager(m_llvm_jitmm)));
 
-    engine_builder.setOptLevel (llvm::CodeGenOpt::Default);
-
-    /* Future site of target options setting */
+    engine_builder.setOptLevel (jit_aggressive()
+                                ? llvm::CodeGenOpt::Aggressive
+                                : llvm::CodeGenOpt::Default);
+
+    llvm::TargetOptions options;
+    // Enables FMA's in IR generation.
+    // However cpu feature set may or may not support FMA's independently
+    options.AllowFPOpFusion = jit_fma() ? llvm::FPOpFusion::Fast :
+                                          llvm::FPOpFusion::Standard;
```

Comment:
Previous default of target options defaults to  llvm::FPOpFusion::Standard // Only allow fusion of 'blessed' ops (currently just fmuladd). 
To truly disable FMA, might need to replace llvm::FPOpFusion::Standard 
with llvm::FPOpFusion::Strict // Never fuse FP-ops.

Reply:
I will simply note this as a comment.